xds/internal/xdsclient/transport/ads/ads_stream.go

```diff
@@ -660,9 +676,11 @@ func (s *Stream) startWatchTimersLocked(typ xdsresource.Type, names []string) {
 }
 
 func resourceNames(m map[string]*ResourceWatchState) []string {
-	ret := make([]string, 0, len(m))
-	for i := range m {
-		ret = append(ret, i)
+	ret := make([]string, len(m))
+	idx := 0
+	for name := range m {
+		ret[idx] = name
+		idx++
```

Comment:
Optional: could do for i, name := range m and use i as index instead.

Reply:
`m` is a map. `range` returns the key and value and not the index.

Reply:
Oh sorry right.